### ISSUE TITLE
Passing test cases need to meet the accessibility requirement

### ISSUE THREAD

**Issue by WilcoFiers** [act-rules/act-rules.github.io#546]
As part of #486, I've worked on creating a clearer definition of what an implementation is. One problem I came across when looking at this is that some of our test cases don't work if I set myself the following requirement:

> If `rule A` is a superset of `rule B`, and `implementation A` is an implementation of `rule A`, than `implementation A` is also an implementation of `rule B`.

This is important, because it makes the scope of a test irrelevant to whether or not a rule has been implemented. This is particularly common with expert evaluations. An export does not test whether or not a page has a language attribute independent of them testing if the language attribute exists, or if it is correct. More often than not this is captured in a single test procedure.

Our test cases today don't work like that. For example, a `pass` test case for "HTML has lang attribute" is this:

```html
<html xml:lang="xyz"></html>
```

Even though this test case passes the rule, it fails the accessibility requirement. I think these kinds of test cases need to be removed, possibly inverted as failures, or moved to other rules entirely. More generally, all `passed` and `inapplicable` test cases have to satisfy the accessibility requirements.

Additionally, I think we need to change some of the `failed` test cases so that the only thing in the example that doesn't satisfy the accessibility requirement is something that also fails the rule. For example the following failed example for "ARIA state or property has valid value":

```html
<div role="button" aria-expanded="mixed"></div>
```

Has to be changed to give the button an accessible name:

```html
<div role="button" aria-expanded="mixed">Expand</div>
```

### Pros

1. This makes implementations about whether or not an implementor has the same interpretation of the accessibility requirement, rather than it being about having the exact same scope in it rules.

2. It means adopting rules could be far less work. Rather than having to implement all new rules, more likely is that existing rules can be updated.

3. It makes for better examples, as we are never saying something "passed", that doesn't satisfy the underlying accessibility requirement

### Cons

1. The test cases will be more complex, and less atomic

2. There are fewer guarantees that an implementor isn't doing "additional" testing in its implementation. This can partially be mitigated by adding more failure test cases.

**Comment by Jym77** [comment]
I agree that giving "wrong" examples is not really a good idea. In addition to implementations, we can have developers looking at the ACT-Rules (*e.g.* because their tool spotted a failure and pointed here for documentation) and believing that what they see here is "the thing to do" even if it's not.

On the other hand, there is also the risk that people start to think that rules do more than what they actually do, and that passing the rule is enough to pass the SC. While it is clearly stated otherwise (in the requirements), it may still be a god idea to have an example pointing this out. "Beware! this rule checks for presence of a title but you still need your title to actually describe the content, look the rule accepts a title of ’etiuntsreuirna’ which is obviously wrong" can also be a good reminder to (human) consumers of the rules.

**Comment by EmmaJP** [comment]
If there are other related rules, including for example one that checks that the title is descriptive, it should be clear in some way that a rule is only part of what is needed for the SC.

A composite rule will do that for most situations. I think I saw something about atomic rules providing info on what composite rules they are used in, when applicable. If that is happening, then the "Beware" only needs to point out that the rule is part of a collection of atomic rules for the SC.
